endpoints-framework/src/test/java/com/google/api/server/spi/discovery/DiscoveryGeneratorTest.java

```diff
@@ -110,6 +116,57 @@ public void testWriteDiscovery_ArrayEndpoint() throws Exception {
     compareDiscovery(expected, doc);
   }
 
+  @Test
+  public void testWriteDiscovery_MapEndpoint() throws Exception {
+    RestDescription doc = getDiscovery(new DiscoveryContext(), MapEndpoint.class);
+    RestDescription expected = readExpectedAsDiscovery("map_endpoint.json");
+    compareDiscovery(expected, doc);
+  }
+
+  @Test
+  public void testWriteDiscovery_MapEndpoint_Legacy() throws Exception {
+    System.setProperty(FORCE_JSON_MAP_SCHEMA.systemPropertyName, "");
+    try {
+      RestDescription doc = getDiscovery(new DiscoveryContext(), MapEndpoint.class);
+      RestDescription expected = readExpectedAsDiscovery("map_endpoint_legacy.json");
+      compareDiscovery(expected, doc);
+    } finally {
+      System.clearProperty(FORCE_JSON_MAP_SCHEMA.systemPropertyName);
+    }
+  }
+
+  @Test
+  public void testWriteDiscovery_MapEndpoint_InvalidKeyTypeKO() throws Exception {
```

Comment:
Is "KO" intentional?

Reply:
It is, but I will make the name more explicit.